File: frontend/src/app/core/models/note.ts
import { TModel } from '@core/models/util-types';
import { ITag } from '@core/models/tag';

export interface INote extends TModel {
  id: number;
  title: string;
  content: string;
  userId: string;
  createdAt: string;
  updatedAt: string;
  reminder: string | null;
  tags: ITag[];
}

File: backend/Models/Note.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace NotesBackend.Models
{
    public class Note
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public string UserId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public User? User { get; set; }
        public int ReminderId { get; set; }
        public Reminder? Reminder { get; set; }
        public List<NoteTag>? NoteTags { get; set; }
    }
}
